P2406-counted-iterator-and-input-iterators.bs

```diff
@@ -262,6 +267,101 @@ performance, can choose using `take` when they know it works correctly for their
 case.
 
 
+# Design alternative
+
+## Motivation
+
+LEWG reviewed R4 of this paper (Issaquah - 2023-02-07 morning) with the
+following poll results:
+
+### Poll outcomes
+
+#### POLL: Reject the NB Comment US-46-107
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 3 <td> 8 <td> 9 <td> 2 <td> 2
+</table>
+
+Outcom: No consensus
+
+WA: Preference to break C++20 sooner
+
+#### POLL: Change `counted_iterator` to have the proposed behavior of `lazy_counted_iterator`, which is a breaking change.
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 6 <td> 12 <td> 2 <td> 1 <td> 3
+</table>
+
+Outcome:
+
+#### POLL: Change iterator category to at most `forward`
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 0 <td> 8 <td> 7 <td> 0 <td> 2
+</table>
+
+Outcome:
+
+#### POLL: Require `forward` as a minimum underlying iterator category
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 0 <td> 3 <td> 5 <td> 3 <td> 3
+</table>
+
+Outcome:
+
+#### POLL: Remove `base` from `lazy_counted_iterator`
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 1 <td> 9 <td> 1 <td> 3 <td> 1
+</table>
+
+Outcome:
+
+
+## Alternative 1: Replace `counted_iterator` with `lazy_counted_iterator`
+
+This alternative propse to change `counted_iterator` to be what this paper
+suggests as `lazy_counted_iterator` or a variation of it.
+
+Issues with this alternative are:
+
+1. This is breaking change (both ABI and source breaking, and depending on the
+   design it might be a silent break of behavior change too)
+2. At least one implementer can't/won't ship it
```

Comment:
We can add additional context here: Another implementer (Jonathan Wakely) could ship it but seems to vaguely prefer not to and is wondering whether or not breaking this now is worth it.